### PR TITLE
Scope policyset results to placement

### PR DESCRIPTION
**Pull request by willkutler**
Fixes an issue where the policyset results field was not scoped to the policyset placement

### CODE REVIEW

File: controllers/propagator/propagation.go [hunk 1]
```diff
@@ -745,7 +745,7 @@ func getClusterPlacementDecisions(
 }
 
 // getPlacementDecisions gets the PlacementDecisions for a PlacementBinding
-func getPlacementDecisions(c client.Client, pb policiesv1.PlacementBinding,
+func GetPlacementDecisions(c client.Client, pb policiesv1.PlacementBinding,
```
Comment (ycao56) [inline controllers/propagator/propagation.go:748]: can you move this to common instead?

File: controllers/policyset/policyset_controller.go [hunk 2]
```diff
@@ -159,6 +146,46 @@ func processPolicySet(ctx context.Context, c client.Client, plcSet *policyv1.Pol
 					placementsByBinding[placement.PlacementBinding] = plcPlacementToSetPlacement(*placement)
 				}
 			}
+
+			// create list of all relevant clusters
+			clusters := []string{}
+			for pbName := range placementsByBinding {
```
Comment (ycao56) [inline controllers/policyset/policyset_controller.go:152]: You already have the placement info in `placementsByBinding` map. Isn't it easier to query the placementDecision and placementrule to get a list of applicable clusters and then filter out the policy.status to generate the results?

Comment (ycao56) [reply controllers/policyset/policyset_controller.go:152]: `GetPlacementDecisions` function generates `placements` which you don't really need here and you have to filter them again.

Comment (willkutler) [reply controllers/policyset/policyset_controller.go:152]: I thought placementDecision didn't show up in policy status?

Comment (ycao56) [reply controllers/policyset/policyset_controller.go:152]: You need to query `placementDecision`and `placementrule` in order know which cluster needs to apply. That's what `GetPlacementDecisions` does. However, this function also process and return a list of placements which you don't need and you didn't use it. It's better to refactor the function to return `placements` and `decisions` separately.

File: controllers/propagator/propagation.go [hunk 3]
```diff
@@ -714,6 +740,27 @@ func getClusterPlacementDecisions(
 		}
 	}
 
+	return placements, nil
+}
+
+// getClusterPlacementDecisions return the placement decisions from cluster
+// placement decisions
+func GetClusterPlacementDecisions(
```
Comment (ycao56) [inline controllers/propagator/propagation.go:748]: suggest to move to `common`

**Comment by ycao56** [comment]
@willkutler can you squash the commits and update the comment in commit and pr?

File: controllers/common/common.go [hunk 4]
```diff
@@ -107,3 +113,79 @@ func GetRetryOptions(logger logr.Logger, retryMsg string, attempts uint) []retry
 		retry.LastErrorOnly(true),
 	}
 }
+
+// getClusterPlacementDecisions return the placement decisions from cluster
```
Comment (ycao56) [inline controllers/common/common.go:117]: `G`etClusterPlacementDecisions

File: controllers/common/common.go [hunk 5]
```diff
@@ -107,3 +113,79 @@ func GetRetryOptions(logger logr.Logger, retryMsg string, attempts uint) []retry
...
+			decisions = append(decisions, *decided)
+		}
+	}
+
+	return decisions, nil
+}
+
+// getApplicationPlacementDecisions return the placement decisions from an application
+// lifecycle placementrule
+func GetApplicationPlacementDecisions(
```
Comment (ycao56) [inline controllers/common/common.go:168]: `G`etApplicationPlacementDecisions

**Review by ycao56 — APPROVED** [review]

**Pull request merged**